src/cbackend.cpp

```diff
@@ -5517,61 +5111,36 @@ class MaskOpsCleanupPass : public llvm::BasicBlockPass {
         // Declare the __not, __and_not1, and __and_not2 functions that we
         // expect the target to end up providing.
         notFunc =
-#if ISPC_LLVM_VERSION <= ISPC_LLVM_4_0
-            llvm::dyn_cast<llvm::Function>(m->getOrInsertFunction("__not", mt, mt, NULL));
-#elif ISPC_LLVM_VERSION <= ISPC_LLVM_8_0 // LLVM 5.0-LLVM 8.0
+#if ISPC_LLVM_VERSION <= ISPC_LLVM_8_0 // LLVM 6.0-LLVM 8.0
             llvm::dyn_cast<llvm::Function>(m->getOrInsertFunction("__not", mt, mt));
 #else                                    // LLVM 9.0+
             llvm::dyn_cast<llvm::Function>(m->getOrInsertFunction("__not", mt, mt).getCallee());
 #endif
         assert(notFunc != NULL);
-#if ISPC_LLVM_VERSION == ISPC_LLVM_3_2
-        notFunc->addFnAttr(llvm::Attributes::NoUnwind);
-        notFunc->addFnAttr(llvm::Attributes::ReadNone);
-#else /* LLVM 3.3+ */
         notFunc->addFnAttr(llvm::Attribute::NoUnwind);
         notFunc->addFnAttr(llvm::Attribute::ReadNone);
-#endif
 
         andNotFuncs[0] =
-#if ISPC_LLVM_VERSION <= ISPC_LLVM_4_0
-            llvm::dyn_cast<llvm::Function>(m->getOrInsertFunction("__and_not1", mt, mt, mt, NULL));
-#elif ISPC_LLVM_VERSION <= ISPC_LLVM_8_0 // LLVM 5.0-LLVM 8.0
+#if ISPC_LLVM_VERSION <= ISPC_LLVM_8_0 // LLVM 6.0-LLVM 8.0
```

Comment:
I think "6.0-8.0" comment is not relevant anymore

Reply:
Yes, it's 6.0-8.0, given that we support only 6.0+, but my point is that this comment is not informative anymore and probably it's better to remove. I.e. it's a binary choice now between "before 8.0" and "8.0+".